### PR TITLE
[MINOR] : Code Cleanup - Misc modules

### PR DESCRIPTION
**Pull request by sjhajharia**
## What
Code Cleanup in Misc Modules

## Changes
Some common changes include

- Replace the Arrays.asList() with Collections.singletonList() wherever possible
- Cleaning up some if-else blocks
- Replacing some instances of String.builder() with String operations

Thanks!

### CODE REVIEW

**Comment by chia7712** [comment]
@sjhajharia Could you please fix the conflicts

**Comment by sjhajharia** [comment]
Thanks @chia7712 !
Just merged the trunk and resolved the conflicts. Hopefully it gets reviewed this time before ti turns stale again! 😅

File: core/src/test/java/kafka/server/handlers/DescribeTopicPartitionsRequestHandlerTest.java [hunk 1]
```diff
@@ -105,12 +106,12 @@ public KafkaPrincipal deserialize(byte[] bytes) throws SerializationException {
...
         .setId(0)
         .setRack("rack")
-        .setEndpoints(Arrays.asList(
-            new UpdateMetadataRequestData.UpdateMetadataEndpoint()
-                .setHost("broker0")
-                .setPort(9092)
-                .setSecurityProtocol(SecurityProtocol.PLAINTEXT.id)
-                .setListener(plaintextListener.value())
+        .setEndpoints(Collections.singletonList(
+                new UpdateMetadataRequestData.UpdateMetadataEndpoint()
```
Comment (chia7712) [inline core/src/test/java/kafka/server/handlers/DescribeTopicPartitionsRequestHandlerTest.java:110]: Could you please revert those unrelated changes to reduce the size of PR?

**Review by chia7712 — COMMENTED** [review]
@sjhajharia thanks for this cleanup. overall LGTM
